modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -98,7 +105,8 @@ class ONNXCompiled {
     std::size_t numInputs() const { return params.num_in; }
     std::size_t numOutputs() const { return params.num_out; }
     void setInput(int i, const cv::Mat &m);
-    void setOutput(int i, cv::Mat &m);
+    void setInput(ONNXCallContext &ctx, const int in_idx, const int name_idx, Views &views, const cv::Rect &roi);
```

Comment:
This overload requires much more parameters than the old one. What's the difference?

Reply:
We get Mat / MediaFrame(create new View and put it in `views`) from `context`. And we call function with `roi` parameter when using InferROI/List/List2. Checking type of infer operation in setInput() more difficult, than simple get from InferROI/List/List2's run().

in_idx and name_idx: 
- in_idx - CallContext iterator,
- name_idx - input names iterator (for tensor info, ...).

Why?
- InferList  - `infer(rois, in)`. `roi` has zero position and we hardcoded `ctx.inMat(1)` ( `setInput( .., 1, 0, ..)` );
- InferROI - `infer(roi, in)`. same as in InferList.
- InferList2 - `infer2(in, roi)`. `in` has zero position and `name_idx` same `in_idx` ( `setInput( .., idx, idx, ..)` );
- Infer - `infer(in)`. same as in inferList2.

`setInput(...)` is more `extractBlob() from IE backend` than old one.
`void setInput(int i, const cv::Mat &m)` is now used for the second infer2 blob-parameter (one case), because we "assume Mats are always BLOBS (not images)" and this input not MediaFrrame.

Reply:
Logic is split. Added `extractMat` instead `setInput(with new parameters)`. Use old version of `setInput` now.
`extractMat()` - extracts cv::Mat from input cv::MediaFrame/cv::Mat.